### ISSUE TITLE
Lower contrast CSS option

### ISSUE THREAD

**Issue by acrymble** [programminghistorian/jekyll#1524]
I had a request today at a workshop for an alternate lower contrast css file for people who find white and black too hard on their eyes. If this is relatively easy it would be a good accessibility option.

**Comment by acrymble** [comment]
it looks like this tutorial walks through the process. It uses javascript

https://dev.to/ananyaneogi/create-a-dark-light-mode-switch-with-css-variables-34l8

**Comment by walshbr** [comment]
I am not sure exactly how to implement this, but it does seem like a good idea.

**Comment by mdlincoln** [comment]
It is a good idea, however I don't really have the CSS chops to implement this either. Can poll the rest of @programminghistorian/technical-team at our upcoming call to see if there's capaicty to take this on.

**Comment by mdlincoln** [comment]
I'm closing this out for now and adding it to our long-term feature request backlog, as we do not have the current skillset to implement this on top of our other high-complexity priorities. Will re-assess in 6 months.

https://github.com/programminghistorian/jekyll/wiki/Technical-Team#new-feature-backlog

**Comment by walshbr** [comment]
@ZoeLeBlanc I took an initial pass at this issue and pushed it to the contrast-mode branch. If you could pull it down and take a look I would appreciate it. Right now it's a button that sits in the nav bar (which is already too crowded to have something like this fit there, I think…

Some things I think I need thoughts on - 

- Where should the button sit?
- Is it doing everything we want?
- How does the code need to be refactored?
- Is it an issue that this version of it accesses the user's browser state to remember their previous selection?

**Comment by acrymble** [comment]
@walshbr does the last item mean a cookie or similar? If so we have to add it to our privacy policy.

**Comment by walshbr** [comment]
I don't actually know the distinction to be able to say @acrymble. @ZoeLeBlanc would probably know? We can also disable that.

**Comment by ZoeLeBlanc** [comment]
Just pushed up a few tweaks to your branch. I tried to get night mode working with url parameters but it causes too many issues with in-lesson links (we would have to reload constantly). So instead I implemented this with sessionStorage, which does persist within a tab, but not between tabs or browser sessions (which localStorage does). If we decide to go forward with this, we'll definitely need to update the privacy policy.

For updating the nav bar, I think we could experiment with adding a smaller button (maybe with the crescent moon icon) in the nav so that it doesn't take up so much room. I wonder if we also want to try out making the nav bar sticky or adding a scroll to top button? Right now it's kind of a pain on the lesson pages to get back up to the top. Another option to try is having the night mode button float on the side. Final question is if we want to infer users location to activate night mode automatically? 

In working on this, I realized we could get rid of URI.min.js and just use built-in URL functionality so I might once this is done redo some of the lessonfilter.js.

Thanks for all your hard work on this @walshbr 👏 🎉

**Comment by walshbr** [comment]
I can look at the nav bar design. @acrymble and @ZoeLeBlanc can you confirm that the way we're doing this is allowed by our privacy policy first? Or that we could at least update our privacy to allow it.

Inferring night mode seems a little unnecessarily slick, personally. I dig the idea, but I'm not sure we need to worry about it.

**Comment by acrymble** [comment]
Sorry @walshbr that's beyond my expertise. If it stores a cookie we need to update the privacy policy to describe what the cookie does (e.g. remembers your preference so you get the same contrast experience when you visit in future - or whatever). I think that can be updated at the very end after you have a solution you're happy with.

**Comment by walshbr** [comment]
@acrymble I think my question (the piece that was directed to you at least) was more whether or not our privacy policy is bound by any kind UK legal restrictions (I was thinking of how we had to take a number of actions to try to comply with Plan S). Or whether the privacy policy is something we're wholly able to adjust as our practices change. It sounds like it might just be the latter? If we can just update it with whatever our current usage is, then that seems fine.

**Comment by acrymble** [comment]
@walshbr it's linked to GDPR, which is EU legislation on privacy and transparency of data collection. As of 31 December 2020, the UK will have 'taken back control' so I don't have any idea if the requirements will remain in force. But maybe it's best for us to err on the side of caution and assume so until we hear otherwise.

**Comment by ZoeLeBlanc** [comment]
I'm not a GDPR expert but I think in this case we aren't persisting any of the data in a database, nor are we even persisting it across browser sessions so essentially it would be almost impossible to do anything with the data in sessionStorage that would infringe on privacy beyond knowing a person's location... which is already a problem if you use a modern web browser or really any ip address 😅 .

I think documenting it is helpful and we might want to consider if we want to use other URL params as sessionStorage tokens instead (we use this for sorting and search of lessons), but I think the potential for data privacy problems is relatively small (especially compared to google analytics for example).

**Comment by walshbr** [comment]
K - I'll take a look at the design aspects of this after break and try to see if I can get a viable version of it working.

**Comment by ZoeLeBlanc** [comment]
If you need any help, feel free to ping me @walshbr and thanks for taking this on 😄

**Comment by ZoeLeBlanc** [comment]
from our #1941 meeting we also need to check if we're conforming to best accessibility practices. @walshbr smartly pointed out checking if font awesome has docs to this effect.
